homeassistant/components/emulated_hue/config.py

```diff
@@ -207,25 +207,25 @@ def get_entity_name(self, state: State) -> str:
         return state.attributes.get(ATTR_EMULATED_HUE_NAME, state.name)  # type: ignore[no-any-return]
 
     @cache  # pylint: disable=method-cache-max-size-none
-    def get_exposed_states(self) -> list[State]:
+    def get_exposed_entity_ids(self) -> list[str]:
         """Return a list of exposed states."""
         state_machine = self.hass.states
         if self.expose_by_default:
             return [
-                state
+                state.entity_id
                 for state in state_machine.async_all()
                 if self.is_state_exposed(state)
             ]
-        states: list[State] = []
-        for entity_id in self.entities:
-            if (state := state_machine.get(entity_id)) and self.is_state_exposed(state):
-                states.append(state)
-        return states
+        return [
+            entity_id
+            for entity_id in self.entities
+            if (state := state_machine.get(entity_id)) and self.is_state_exposed(state)
```

Comment:
We're caching something here that relies on he state machine. I know that's existing code but that seems weird ?

Reply:
The cache gets cleared in the function below it when needed